pallets/evm-contract-helpers/src/eth.rs

```diff
@@ -62,10 +80,28 @@ impl<T: Config> ContractHelpers<T> {
 		Ok(())
 	}
 
+	/// Get contract sponsoring mode
+	/// @param contract_address Contract to get sponsoring mode of
+	/// @return uint8 Current sponsoring mode
+	///  0 for disabled
+	///  1 for allowlisted
+	///  2 for generous
+	fn sponsoring_mode(&self, contract_address: address) -> Result<uint8> {
+		Ok(<Pallet<T>>::sponsoring_mode(contract_address).to_eth())
+	}
+
+	/// Set contract sponsoring mode
+	/// @param contract_address Contract to change sponsoring mode of
+	/// @param mode Chosen sponsoring mode
+	///  0 for disabled
+	///  1 for allowlisted
+	///  2 for generous
+	/// @dev Only contract owner may change this setting
 	fn set_sponsoring_mode(
 		&mut self,
 		caller: caller,
 		contract_address: address,
+		// TODO: implement support for enums in evm-coder
```

Comment:
How would enums be supported in Solidity?

Reply:
Solidity supports enums
```solidity
enum SponsoringMode { ALLOWLIST, GENEROUS }
```

Currently, evm-coder doesn't support generation of such enums, and we accept enum-like values as plain u8, so instead of

```solidity
setSponsoringMode(contract, SponsoringMode.GENEROUS)
```

User should write

```solidity
setSponsoringMode(contract, 1)
```

And this is ugly

I see enum support implementation as something like

```rust
#[derive(SolidityEnum)]
#[repr(u8)]
enum SponsoringMode {
  Allowlist,
  Generous,
}

// In #[solidity_interface] definition
fn set_sponsoring_mode(&mut self, #[enum] mode: SponsoringMode)
```

But currently, we have no plans on implementing this